### PR TITLE
docs: update why-playwright.md

### PR DESCRIPTION
**Pull request by arjunattam**
* Fixes errors in previous version
* Adds new section on limitations covering IE11, Java bindings and testing on real mobile devices

### CODE REVIEW

File: docs/why-playwright.md [hunk 1]
```diff
@@ -41,8 +42,16 @@ Playwright enables fast, reliable and capable automation across all modern brows
...
 
 * **Language bindings**. Playwright is also available in [Python](https://github.com/microsoft/playwright-python) and [C#](https://github.com/hardkoded/playwright-sharp).
 
 * **Deploy tests to CI**. First-party [Docker image](docker/README.md) and [GitHub Actions](https://github.com/microsoft/playwright-github-action) to deploy tests to [your preferred CI/CD provider](ci.md).
+
+## Limitations
+
+* **Legacy Edge and IE11 support**. Playwright does not support legacy Microsoft Edge or IE11 ([deprecation notice](https://techcommunity.microsoft.com/t5/microsoft-365-blog/microsoft-365-apps-say-farewell-to-internet-explorer-11-and/ba-p/1591666)). The new Microsoft Edge (on Chromium) is supported.
+
+* **Java language bindings**: The Playwright API cannot be used in Java or Ruby today. This is a temporary limitation as Playwright is built to support non-JavaScript language bindings.
```
Comment (yury-s) [inline docs/why-playwright.md:55]: > ... to support non-JavaScript language bindings.

Maybe rephrase it as 'any language' I was confused a little bit first time I read.

**Review by yury-s — APPROVED** [review]

File: docs/why-playwright.md [hunk 2]
```diff
@@ -16,7 +17,7 @@ Playwright enables fast, reliable and capable automation across all modern brows
 
 * **Test for mobile**. Use [device emulation](emulation.md) to test your responsive web apps in mobile web browsers.
 
-* **Headless and headful**. Headless and headful automation is supported across all browsers and on all platforms.
+* **Headless and headful**. Playwright supports headless (without browser UI) and headful (with browser UI) modes for all browsers and all platforms. Headful is great for debugging, and headless is faster.
```
Comment (yury-s) [inline docs/why-playwright.md:20]: > and headless is faster.

Maybe mention CI/cloud?

File: docs/why-playwright.md [hunk 3]
```diff
@@ -41,8 +42,16 @@ Playwright enables fast, reliable and capable automation across all modern brows
...
 
 * **Deploy tests to CI**. First-party [Docker image](docker/README.md) and [GitHub Actions](https://github.com/microsoft/playwright-github-action) to deploy tests to [your preferred CI/CD provider](ci.md).
+
+## Limitations
+
+* **Legacy Edge and IE11 support**. Playwright does not support legacy Microsoft Edge or IE11 ([deprecation notice](https://techcommunity.microsoft.com/t5/microsoft-365-blog/microsoft-365-apps-say-farewell-to-internet-explorer-11-and/ba-p/1591666)). The new Microsoft Edge (on Chromium) is supported.
+
+* **Java language bindings**: The Playwright API cannot be used in Java or Ruby today. This is a temporary limitation as Playwright is built to support non-JavaScript language bindings.
+
+* **Test on real mobile devices**: Playwright runs against desktop browsers and supports mobile emulation. If you are interested in this, please [upvote this issue](https://github.com/microsoft/playwright/issues/1122).
```
Comment (dgozman) [inline docs/why-playwright.md:57]: Suggest the following:

```
Playwright uses desktop browsers to emulate mobile devices. If you are interested in running on the real mobile devices, please upvote this issue.
```

**Review by dgozman — APPROVED** [review]
